docs/src/getting-started/0-47-0-migration-guide.md

```diff
@@ -0,0 +1,56 @@
+# `snforge` 0.47.0 Migration Guide
+
+Starting from version 0.47.0, `snforge` will by default support Scarb versions 2.12 or newer.
+This is due by migration to the Scarb V2 version of procedural macros, which are used to handle arguments like `#[test]`
+in `snforge`.
+The benefit of this migration is that tools like Cairo plugin to VSCode will start showing better, more descriptive
+errors.
+
+To continue using `snforge` you will need to perform different actions depending on your Scarb version.
+
+## Scarb Versions >= 2.12.0
+
+For Scarb versions >= 2.12.0, we recommend upgrading your `snforge_std` dependency to the one matching your `snforge`
+installation (0.47.0 at the time of writing of this doc).
+
+In your `Scarb.toml` file, update the dependency.
+
+```toml
+[dev-dependencies]
+snforge_std = "0.47.0"
+```
+
+No further action is required.
+
+## Scarb Versions < 2.12.0
+
+> 📝 **Note**
+>
+> We recommend upgrading to at least Scarb 2.12.0, as steps outlined below will stop being supported soon.
+
+For Scarb versions < 2.12.0 it is still possible to continue using the latest `snforge`.
+We now publish a new package `snforge_std_compatibility`, supporting versions < 2.12.0.
+
+> 📝 **Note**
+>
+> `snforge_std_compatibility` may not provide full functionality in comparison to `snforge_std`.
+
+First, in your `Scarb.toml`, remove the `snforge_std` dependency and add `snforge_std_compatibility`.
+
+```diff
+[dev-dependencies]
+- snforge_std = "0.46.0"
++ snforge_std_compatibility = "0.47.0"
+```
+
+Next, replace all kinds of imports in your code from `snforge_std` to `snforge_std_compatibility`
+
+```diff
+// Replace use statements
+- use snforge_std::{ContractClassTrait, start_cheat_caller_address};
++ use snforge_std_compatibility::{ContractClassTrait, start_cheat_caller_address};
+
+// Replace full path usages
+- let result = snforge_std::declare("MyContract").unwrap();
++ let result = snforge_std_compatibility::declare("MyContract").unwrap();
+```
```

Comment:
I would somehow involve a table here, so it's compared left vs right. It seems more natural for me when seeing comparison.

Reply:
Codeblocks with `diff` format generate nice, colored output.

<img width="818" height="98" alt="obraz" src="https://github.com/user-attachments/assets/ed0992ad-9d7e-4bb4-815a-01814eb0e2f9" />

With the table we would to somehow have to style it ourselves because they don't seem to render codeblocks.

Reply:
Ok let's leave it like that then.